src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -0,0 +1,285 @@
+    }
+
+    internal sealed class GeneralCommandErrorFeedback : IFeedbackProvider
+    {
+        private readonly Guid _guid;
+        private readonly object[] _args;
+        private ScriptBlock? _fuzzySb;
+
+        internal GeneralCommandErrorFeedback()
+        {
+            _guid = new Guid("A3C6B07E-4A89-40C9-8BE6-2A9AAD2786A4");
+            _args = new object[1];
+        }
+
+        public Guid Id => _guid;
+
+        public string Name => "General";
+
+        public string Description => "The built-in general feedback source for command errors.";
+
+        public string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token)
+        {
+            var rsToUse = Runspace.DefaultRunspace;
+            if (rsToUse is null)
+            {
+                return null;
+            }
+
+            if (lastError.FullyQualifiedErrorId == "CommandNotFoundException")
+            {
+                EngineIntrinsics context = rsToUse.ExecutionContext.EngineIntrinsics;
+
+                var target = (string)lastError.TargetObject;
+                CommandInvocationIntrinsics invocation = context.SessionState.InvokeCommand;
+
+                // See if target is actually an executable file in current directory.
+                var localTarget = Path.Combine(".", target);
+                var command = invocation.GetCommand(
+                    localTarget,
+                    CommandTypes.Application | CommandTypes.ExternalScript);
+
+                if (command is not null)
+                {
+                    return StringUtil.Format(
+                        SuggestionStrings.Suggestion_CommandExistsInCurrentDirectory,
+                        target,
+                        localTarget);
+                }
+
+                // Check fuzzy matching command names.
+                if (ExperimentalFeature.IsEnabled("PSCommandNotFoundSuggestion"))
+                {
+                    _fuzzySb ??= ScriptBlock.CreateDelayParsedScriptBlock(@$"
+                        param([string] $target)
+                        $cmdNames = Get-Command $target -UseFuzzyMatch | Select-Object -First 10 -ExpandProperty Name
+                        if ($cmdNames) {{
+                            [string]::Join(', ', $cmdNames)
+                        }}
+                    ", isProductCode: true);
+
+                    _args[0] = target;
+                    var result = _fuzzySb.InvokeReturnAsIs(_args);
+
+                    if (result is not null && result != AutomationNull.Value)
+                    {
+                        return StringUtil.Format(
+                            SuggestionStrings.Suggestion_CommandNotFound,
+                            result.ToString());
+                    }
+                }
+            }
+
+            return null;
+        }
+    }
+
+    internal sealed class UnixCommandNotFound : IFeedbackProvider, ICommandPredictor
```

Comment:
@SteveL-MSFT @JamesWTruher @theJasonHelmick Need your thoughts on whether this one should be in S.M.A. It certainly can be separated out as a module, but seems to be an overkill, and also there is discovery issue.

Reply:
It may be better to have this as a separate module to serve as an example.  I expect WinGet and PSGet to implement feedback providers.

Reply:
I will keep the code here for now, but it's made no longer auto registered.
Once I have the changes ready in UnixCompleter module, I will remove this class from S.M.A.